cpp/src/dictionary/remove_keys.cu

```diff
@@ -57,38 +58,42 @@ std::unique_ptr<column> remove_keys_fn(
   // create keys positions column to identify original key positions after removing they keys
   auto const keys_view = dictionary_column.keys();
   auto execpol         = rmm::exec_policy(stream);
-  rmm::device_vector<int32_t> keys_positions(keys_view.size());  // needed for remapping indices
+  rmm::device_uvector<uint32_t> keys_positions(keys_view.size(), stream);  // for remapping indices
   thrust::sequence(execpol->on(stream), keys_positions.begin(), keys_positions.end());
   column_view keys_positions_view(
-    data_type{type_id::INT32}, keys_view.size(), keys_positions.data().get());
+    data_type{type_id::UINT32}, keys_view.size(), keys_positions.data());
 
   // copy the non-removed keys ( keys_to_keep_fn(idx)==true )
-  rmm::device_vector<int32_t> map_indices(keys_view.size(), -1);  // init -1 to identify new nulls
+  // init to max identify new nulls
+  rmm::device_uvector<uint32_t> map_indices(keys_view.size(), stream);
+  thrust::fill(execpol->on(stream),
+               map_indices.begin(),
+               map_indices.end(),
+               static_cast<uint32_t>(std::numeric_limits<size_type>::max()));
```

Comment:
Suggest a comment to make this clear. 

```suggestion
               static_cast<uint32_t>(std::numeric_limits<size_type>::max()));  // note max of `size_type`, not `uint32_t`
```

Reply:
Changed to use `max_column_size` to make it clearer.